db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/UserTypeConstantNode.java

```diff
@@ -54,194 +54,199 @@
-	 *			getValue() returns the DataValueDescriptor for the built-in
-	 *			types that are implemented as user types (date, time, timestamp)
-	 */
-    public	Object	getObjectValue() { return value; }
-
-	/**
-	 * Return whether or not this node represents a typed null constant.
-	 *
-	 */
-	public boolean isNull()
-	{
-		return (value == null);
-	}
-
-	/**
-	 * Return an Object representing the bind time value of this
-	 * expression tree.  If the expression tree does not evaluate to
-	 * a constant at bind time then we return null.
-	 * This is useful for bind time resolution of VTIs.
-	 * RESOLVE: What do we do for primitives?
-	 *
-	 * @return	An Object representing the bind time value of this expression tree.
-	 *			(null if not a bind time constant.)
-	 *
-	 */
-	public Object getConstantValueAsObject()
-	{
-		return value;
-	}
-
-	/**
-	 * For a UserTypeConstantNode, we have to store away the object somewhere
-	 * and have a way to get it back at runtime.
-	 * These objects are serializable.  This gives us at least two options:
-	 * 1) serialize it out into a byte array field, and serialize
-	 *	  it back in when needed, from the field.
-	 * 2) have an array of objects in the prepared statement and a #,
-	 *	  to find the object directly. Because it is serializable, it
-	 *	  will store with the rest of the executable just fine.
-	 * Choice 2 gives better performance -- the ser/deser cost is paid
-	 * on database access for the statement, not for each execution of it.
-	 * However, it requires some infrastructure support from prepared
-	 * statements.  For now, we take choice 3, and make some assumptions
-	 * about available methods on the user type.  This choice has the
-	 * shortcoming that it will not work for arbitrary user types.
-	 * REVISIT and implement choice 2 when a general solution is needed.
-	 * <p>
-	 * A null is generated as a Null value cast to the type of
-	 * the constant node.
-	 *
-	 * @param acb	The ExpressionClassBuilder for the class being built
-	 * @param mb	The method the expression will go into
-	 *
-	 *
-	 * @exception StandardException		Thrown on error
-	 */
-	public void generateExpression(ExpressionClassBuilder acb,
-											MethodBuilder mb)
-									throws StandardException {
-
-		TypeCompiler		tc = getTypeCompiler();
+            {
+                maxWidth = ReuseFactory.getInteger(TypeId.TIMESTAMP_MAXWIDTH);
+                typeId = TypeId.getBuiltInTypeId(Types.TIMESTAMP);
+            }
+            else
+            {
+                if (SanityManager.DEBUG)
+                {
+                    SanityManager.THROWASSERT(
+                            "Unexpected class " + arg1.getClass().getName());
+                }
+            }
+
+            assert maxWidth != null;
+
+            super.init(
+                typeId,
+                (arg1 == null) ? Boolean.TRUE : Boolean.FALSE,
+                maxWidth);
+
+            if( dvd != null) {
+                setValue(dvd);
```

Comment:
With the lines added after this line to set the value, this line become redundant and can be removed.

Reply:
Weirdly enough, no. `setValue` sets the attribute of the base class, whereas `value=` sets the value of this derived class. The whole thing is a bit messy and explained in a comment above `UserTypeConstantNode.value` declaration:

` /*
    ** This value field hides the value in the super-type.  It is here
    ** Because user-type constants work differently from built-in constants.
    ** User-type constant values are stored as Objects, while built-in
    ** constants are stored as StorableDataValues.
    **
    ** RESOLVE: This is a bit of a mess, and should be fixed.  All constants
    ** should be represented the same way.
    */`